Source/UnitTests/Common/MutexTest.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <mutex>
+#include <thread>
+
+#include "Common/Mutex.h"
+
+template <typename MutexType>
+static void DoAtomicMutexTest()
+{
+  MutexType work_mutex;
+
+  bool worker_done = false;
+
+  static constexpr auto SLEEP_TIME = std::chrono::microseconds{1};
+
+  // lock() on main thread, unlock() on worker thread.
+  std::thread thread{[&, lk = std::unique_lock{work_mutex}] {
+    std::this_thread::sleep_for(SLEEP_TIME);
+    worker_done = true;
+  }};
+
+  // lock() waits for the thread to unlock().
+  {
+    std::lock_guard lk{work_mutex};
+    EXPECT_TRUE(worker_done);
+  }
+
+  thread.join();
+
+  // Prevent below workers from incrementing `done_count`.
+  MutexType done_mutex;
+  std::unique_lock done_lk{done_mutex, std::try_to_lock};
+  EXPECT_TRUE(done_lk.owns_lock());
```

Comment:
Isn't it possible for the lock attempt to spuriously fail when using std::try_to_lock?

Reply:
Some/all? of the stdlib mutices allow `try_lock` to spuriously, but this particular mutex implementation will not spuriously fail.